src/backend/utils/misc/queryenvironment.c

```diff
@@ -86,6 +87,7 @@ create_queryEnv2(MemoryContext cxt, bool top_level)
 		queryEnv = topLevelQueryEnv;
 		queryEnv->namedRelList = NIL;
 		queryEnv->dropped_namedRelList = NIL;
+		queryEnv->savedCatcacheMessages = NIL;
```

Comment:
how are we making sure that savedCatcacheMessages is NULL initialized in all cases? do we crate queryEnv with all fields set to NULL?

Reply:
queryEnv is either created with palloc0 through `create_queryEnv`, which sets all fields to null, or here in `create_queryEnv2`, where we specify it explicitly.